app/data_models/questionnaire_store.py

```diff
@@ -120,48 +110,43 @@ def _remove_old_supplementary_lists_and_answers(
         :param new_data - the new supplementary data for comparison
         """
         deleted_list_item_ids: set[str] = set()
-        for list_name, mappings in self.supplementary_data_store.list_lookup.items():
+        for (
+            list_name,
+            mappings,
+        ) in self.data_stores.supplementary_data_store.list_lookup.items():
             if list_name in new_data.get("items", {}):
                 new_identifiers = [
                     item["identifier"] for item in new_data["items"][list_name]
                 ]
                 for identifier, list_item_id in mappings.items():
                     if identifier not in new_identifiers:
-                        self.list_store.delete_list_item(list_name, list_item_id)
+                        self.data_stores.list_store.delete_list_item(
+                            list_name, list_item_id
+                        )
                         deleted_list_item_ids.add(list_item_id)
             else:
-                self.list_store.delete_list(list_name)
+                self.data_stores.list_store.delete_list(list_name)
                 deleted_list_item_ids.update(mappings.values())
-        self.answer_store.remove_all_answers_for_list_item_ids(*deleted_list_item_ids)
+        self.data_stores.answer_store.remove_all_answers_for_list_item_ids(
+            *deleted_list_item_ids
+        )
 
     def _deserialize(self, data: str) -> None:
         json_data = json_loads(data)
-        self.progress_store = ProgressStore(json_data.get("PROGRESS"))
         self.set_metadata(json_data.get("METADATA", {}))
-        self.supplementary_data_store = SupplementaryDataStore.deserialize(
-            json_data.get("SUPPLEMENTARY_DATA", {})
-        )
-        self.answer_store = AnswerStore(json_data.get("ANSWERS"))
-        self.list_store = ListStore.deserialize(json_data.get("LISTS"))
-        self.response_metadata = json_data.get("RESPONSE_METADATA", {})
+        self.data_stores.deserialize_stores(json_data, metadata=self._metadata)
 
     def serialize(self) -> str:
-        data = {
+        serialized_stores = self.data_stores.serialize_stores()
+        data = serialized_stores | {
             "METADATA": self._metadata,
```

Comment:
I see what you mean with the abstraction, though, mixed thoughts, it doesn't seem right to me to have a method thats
```python
def method(self):
    self.data_stores.x.a()
    self.data_stores.y.b()
    self.data_stores.z.c()
```
the entire functionality of it relates to the `data_stores` so I would have thought it makes sense to make the data stores responsible for it. That way if someone comes to change something about a data store, I think they're less likely to accidentally miss logic from another class, even if that does result in a fair bit of it moving.

Reply:
Though I guess this would be fixed by it being a view only add-on though and having each attribute still present on the store, so I think I'd also lean towards the attributes still being in this class